apps/common-app/src/examples/RuntimeTests/ReJest/TestRunner/TrackerRegistry.ts

```diff
@@ -0,0 +1,64 @@
+import { makeMutable, runOnJS } from 'react-native-reanimated';
+import type { TrackerCallCount } from '../types';
+
+let callTrackerRegistryJS: Record<string, number> = {};
+const callTrackerRegistryUI = makeMutable<Record<string, number>>({});
+function callTrackerJS(name: string) {
+  if (!callTrackerRegistryJS[name]) {
+    callTrackerRegistryJS[name] = 0;
+  }
+  callTrackerRegistryJS[name]++;
+}
+
+const notificationRegistry: Record<string, boolean> = {};
+function notifyJS(name: string) {
+  notificationRegistry[name] = true;
+}
+
+export class TrackerRegistry {
+  public notify(name: string) {
+    'worklet';
+    if (_WORKLET) {
+      runOnJS(notifyJS)(name);
+    } else {
+      notifyJS(name);
+    }
+  }
+
+  public async waitForNotify(name: string) {
+    return new Promise(resolve => {
+      const interval = setInterval(() => {
+        if (notificationRegistry[name]) {
+          clearInterval(interval);
+          resolve(true);
+        }
+      }, 10);
+    });
+  }
```

Comment:
You can move it to another class `NotificationManager` or something like that, because this shouldn't be a part of `CallTrackerRegistry`